### PR TITLE
Adds samples for matching stragegy

### LINKED ISSUE #361: Add new code-samples requested by the Meilisearch v0.29 release
⚠️ This issue is generated, it means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Sorry if this is already wholly/partially implemented. Feel free to let me know about the state of this issue in the repo.

Related to https://github.com/meilisearch/integration-guides/issues/213

----

Add the new code samples into the file `.code-samples.meilisearch.yml`:

TODO:
- [ ] Add `search_parameter_guide_matching_strategy_1`
  - Search for `big fat liar` with `"matchingStrategy": "last"` in a `movies` index.
- [ ] Add `search_parameter_guide_matching_strategy_2`
  - Search for `big fat liar` with `"matchingStrategy": "all"` in a `movies` index.

For example, in Javascript, the code-samples required to fix the issue are:

`client.index('movies').search('big fat liar', { matchingStrategy: 'last' })` and also 
`client.index('movies').search('big fat liar', { matchingStrategy: 'all' })`

so the final version inside of the samples should be:

```yml
search_parameter_guide_matching_strategy_1: |-
  client.index('movies').search('big fat liar', { matchingStrategy: 'last' })
search_parameter_guide_matching_strategy_2: |-
  client.index('movies').search('big fat liar', { matchingStrategy: 'all' })
```

### PR DESCRIPTION
**Pull request by romilpunetha**
# Pull Request

## Related issue
Fixes #361

## What does this PR do?
- Adds samples for matching strategy

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by romilpunetha** [comment]
@brunoocasali

**Comment by brunoocasali** [comment]
Hi @romilpunetha! @bidoubiwa is currently on sick leave, and you can expect a review next week :) 

Thanks for contributing! :D

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -599,6 +599,38 @@ search_parameter_guide_show_matches_position_1: |-
...
+  .execute()
+  .await
+  .unwrap();
+  
+  // Get the formatted results
+  let formatted_results: Vec<&Movie> = results
+    .hits
+    .iter()
+    .map(|r| r.formatted_result.as_ref().unwrap())
+    .collect();
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:617]:
```suggestion
```

It's a very nice addition but potentially the user wants the whole object and not only the hits. For consistency with the other examples, lets not add this.

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -599,6 +599,38 @@ search_parameter_guide_show_matches_position_1: |-
...
+  .execute()
+  .await
+  .unwrap();
+  
+  // Get the formatted results
+  let formatted_results: Vec<&Movie> = results
+  .hits
+  .iter()
+  .map(|r| r.formatted_result.as_ref().unwrap())
+  .collect();
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:633]:
```suggestion
```

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -1280,9 +1312,9 @@ delete_a_key_1: |-
 authorization_header_1:
   let client = Client::new("http://localhost:7700", "masterKey");
   let keys = client
-    .get_keys()
-    .await
-    .unwrap();
+  .get_keys()
+  .await
+  .unwrap();
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:1317]: Not sure about this formatting. Is it not more common to indent the methods?

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution 🔥

File: .code-samples.meilisearch.yaml (hunk 3 above)
Comment (romilpunetha) [reply .code-samples.meilisearch.yaml:1305]: I ran `cargo fmt` and the command gave this output. Is there something that I need to change here?

File: .code-samples.meilisearch.yaml (hunk 2 above)
Comment (romilpunetha) [reply .code-samples.meilisearch.yaml]: Fixed this.

File: .code-samples.meilisearch.yaml (hunk 3 above)
Comment (bidoubiwa) [reply .code-samples.meilisearch.yaml:1305]: If cargo fmt said so, lets keep it like that ! 🔥

File: .code-samples.meilisearch.yaml [hunk 4]
```diff
@@ -599,6 +599,26 @@ search_parameter_guide_show_matches_position_1: |-
...
+search_parameter_guide_matching_strategy_2: |-
+  let results: SearchResults<Movie> = client
+  .index('movies')
+  .search()
+  .with_query("big fat liar")
+  .with_matching_strategy(MatchingStrategies::ALL)
+  .execute()
+  .await
+  .unwrap();
+
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:621]:
```suggestion
```

File: .code-samples.meilisearch.yaml [hunk 5]
```diff
@@ -599,6 +599,26 @@ search_parameter_guide_show_matches_position_1: |-
...
+search_parameter_guide_matching_strategy_1: |-
+  let results: SearchResults<Movie> = client
+  .index('movies')
+  .search()
+  .with_query("big fat liar")
+  .with_matching_strategy(MatchingStrategies::LAST)
+  .execute()
+  .await
+  .unwrap();
+
```
Comment (bidoubiwa) [inline .code-samples.meilisearch.yaml:611]:
```suggestion
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

**Review by bidoubiwa — APPROVED** [review]
bors merge

**Pull request merged**